src/Plivo/Resources/Message/MessageInterface.php

```diff
@@ -144,66 +132,101 @@ protected function getList($optionalArgs = [])
-                           array $optionalArgs = [], $powerpackUUID = null)
+    public function __call($name_of_method, $arguments)
     {
-        $mandatoryArgs = [
-            'dst' => implode('<', $dst),
-        ];
+        if ($name_of_method == 'create' and count($arguments) == 1)
+        {
+            $arguments = $arguments[0];
+            $dest = explode('<', $arguments['dst']);
+            if (!array_key_exists('src', $arguments) && !array_key_exists('powerpackUUID', $arguments))
+            {
+                throw new PlivoValidationException("Specify either powerpack_uuid or src in request params to send a message.");
+            }
 
-        if (ArrayOperations::checkNull($mandatoryArgs)) {
-            throw new PlivoValidationException(
-                "Mandatory parameters cannot be null");
-        }
+            if (!array_key_exists('src', $arguments) && array_key_exists('powerpackUUID', $arguments))
+            {
+                throw new PlivoValidationException("Both powerpack_uuid and src cannot be specified. Specify either powerpack_uuid or src in request params to send a message.");
+            }
 
-        if (is_null($src) &&  is_null($powerpackUUID)) {
-            throw new PlivoValidationException(
-                "Specify either powerpack_uuid or src in request params to send a message."
-            );
+            foreach ($dest as $dst)
+            {
+                if (!is_numeric($dst))
+                {
+                    throw new PlivoValidationException('Destination number should be numeric');
+                }
+            }
+            $response = $this
+                ->client
+                ->update($this->uri, $arguments);
+
+            $responseContents = $response->getContent();
+
+            if (!array_key_exists("error", $responseContents))
+            {
+                if (array_key_exists("invalid_number", $responseContents))
+                {
+                    return new MessageCreateResponse($responseContents['message'], $responseContents['message_uuid'], $responseContents['api_id'], $response->getStatusCode() , $responseContents['invalid_number']);
+                }
+                else
+                {
+                    return new MessageCreateResponse($responseContents['message'], $responseContents['message_uuid'], $responseContents['api_id'], $response->getStatusCode() , []);
+                }
+            }
+            else
+            {
+                throw new PlivoResponseException($responseContents['error'], 0, null, $response->getContent() , $response->getStatusCode());
+            }
         }
 
-        if (!is_null($src) && !is_null($powerpackUUID)) {
-            throw new PlivoValidationException(
-                "Both powerpack_uuid and src cannot be specified. Specify either powerpack_uuid or src in request params to send a message."
-            );
-        }
+        else
+        {
+            $src = $arguments[0];
```

Comment:
is there anywhere you mentioned the order for argument?

Reply:
No, this wasn't mentioned earlier as well. Hence coming up with the new interface. This snippet is to handle the old interface